crates/misc/rust/macro/src/lib.rs

```diff
@@ -49,34 +48,50 @@ fn generate_load(item: &syn::ItemTrait) -> syn::Result<TokenStream> {
         #vis fn load_file(path: impl AsRef<std::path::Path>) -> #root::anyhow::Result<#name> {
             let bytes = std::fs::read(path)?;
 
-            let isa = {
-                let isa_builder = #root::cranelift_native::builder()
-                    .map_err(|s| #root::anyhow::format_err!("{}", s))?;
+            use #root::wasmtime_api::{HostRef, Config, Extern, Engine, Store, Instance, Module};
+
+            let config = {
                 let flag_builder = #root::cranelift_codegen::settings::builder();
-                isa_builder.finish(#root::cranelift_codegen::settings::Flags::new(flag_builder))
+                let flags = #root::cranelift_codegen::settings::Flags::new(flag_builder);
+                let features = #root::wasmtime_jit::Features {
+                    multi_value: true,
+                    ..Default::default()
+                };
+                let strategy = #root::wasmtime_jit::CompilationStrategy::Auto;
+                Config::new(flags, features, false, strategy)
             };
+            let engine = HostRef::new(Engine::new(config));
+            let store = HostRef::new(Store::new(&engine));
+            let global_exports = store.borrow().global_exports().clone();
 
-            let mut cx = #root::wasmtime_jit::Context::with_isa(
-                isa,
-                #root::wasmtime_jit::CompilationStrategy::Auto
-            ).with_features(#root::wasmtime_jit::Features {
-                multi_value: true,
-                ..Default::default()
-            });
             let data = #root::wasmtime_interface_types::ModuleData::new(&bytes)?;
+
+            let module = HostRef::new(Module::new(&store, &bytes).expect("module"));
+
+            let mut imports: Vec<Extern> = Vec::new();
             if let Some(module_name) = data.find_wasi_module_name() {
                 let wasi_handle = wasmtime_wasi::instantiate_wasi(
                     "",
-                    cx.get_global_exports(),
+                    global_exports,
                     &[],
                     &[],
                     &[],
                 )?;
-                cx.name_instance(module_name, wasi_handle);
+                let wasi_instance = Instance::from_handle(&store, wasi_handle).expect("wasi instance");
+                for i in module.borrow().imports().iter() {
+                    if i.module().as_str() != module_name {
+                        panic!("unknown import module {}", i.module().as_str());
```

Comment:
This panic now that I see it seems a little weird, is this asserting that every import from a module listed, if any, comes from the wasi module? (if so shouldn't we just skip non-wasi imports here?)